doc/design/scheduled-tasks.md

```diff
@@ -0,0 +1,55 @@
+# Asynchronous Tasks
+We are currently (7/16/2021) using celery to run these asynchronous tasks:
+
+ * Calculating blob sha256 digests
+ * Writing dandiset.yaml and assets.yaml after publishing
+ * Validating asset metadata
+ * Validating dandiset metadata
+
+These all work well at the usage levels that we are currently seeing, but may not scale well into the future.
+Specifically, every time an asset is uploaded, asset metadata and dandiset metadata are re-validated.
+When uploading large batches of assets, dandiset metadata is validated repeatedly and unnecessarily.
+
+## Motivation
+Currently (7/16/2021), every time a Version is saved, it recalculates the value of the `assetsSummary` metadata field.
+This requires performing a `SELECT *` on all Assets in the Version, which does not scale well for Versions with lots of Assets.
+Because uploading a new Asset to a Version involves saving that Version, uploading lots of new assets will eventually start causing request timeouts as the number of Assets increases.
+
+We would like to move this `assetsSummary` calculation into a celery task, but we also don't want to trigger a separate task for every new Asset, as that would only delay the load on the DB.
+We need a way to run this task only once per batch of uploads.
+
+## Proposal
+Instead of adding a new task to the queue whenever something needs to be done, we use a scheduled job that runs the task only when necessary.
+
+Since `assetsSummary` calculation and dandiset metadata validation both need to be triggered in the same circumstances, we'll consider them together.
+
+#### Request thread
+Whenever a Version is modified, a flag is set on the Version indicating that it has been modified (instead of queueing a task as it does now):
+```
+  # Things that modify the version, necessitating a validation/assetsSummary computation
+  ...
+  # validate_version_metadata.delay(version.id)
+  version.status = status.MODIFIED
```

Comment:
there could be many notions of "modified" (well, there is `modified` time stamp which would flip on every change) and necessary actions to be taken.  I wonder if might better become more explicit, e.g.
```
version.assetsSummaryStatus = status.OUTDATED
```
or alike, which would make it cleared that scheduled "updateAssetsSummary" job would be the one to deal with it.